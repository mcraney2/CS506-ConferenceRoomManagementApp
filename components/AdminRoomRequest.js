import axios from 'axios';
import React, { Component } from 'react';
import { SafeAreaView, Text, View } from 'react-native';
import { FlatList } from 'react-native-gesture-handler';
import Button from './Button'

class AdminRoomRequest extends Component {
    constructor(props) {
        super(props);
        this.state = {  }
    }
    acceptRequest(id, refresh) {
        console.log("Send database 'accept' message to add to calendar and then database request to remove this from requests");
        console.log(id);
            //console.log(room,startTime, endTime, currentTime);
        const request = JSON.stringify(
              { 
                adminid:1,
                requestid: id
    
            });
        axios.post('http://10.0.2.2:8000/room_mgmt/admin/requests/process/', request)
            .then(function (response) {
              //console.log(response);
            })
            .catch(function (error) {
              console.log(error);
            })
            .then(function() {
              refresh();
            
  
            });
        }
    
    denyRequest(id, refresh) {
        console.log("Send database 'deny' request and simply remove request");
        console.log(id);
        const request = JSON.stringify(
            { 
              adminid:1,
              requestid: id
  
          });
        axios.put('http://10.0.2.2:8000/room_mgmt/admin/requests/process/',request)
        .then(function (response) {
            //console.log(response);
          })
          .catch(function (error) {
            console.log(error);
          })
          .then(function() {
            refresh();
          

          });
        }
    
    getDate(date) {
          
          
          var year = date.getFullYear();
          var month = date.getMonth() + 1;
          var day = date.getDate();
         
          var total = month + '/' + day + '/' + year;
          return total;
      }
    getTime(date) {
        
        
        var hour = date.getHours();
        var minutes = date.getMinutes();
        if (parseInt(minutes, 10) < 10) {
            minutes = '0' + minutes;
        }
        var total = hour + ':' + minutes;

        return total;
    }
     renderItem = ({ item }) => (
      // start = new Date(this.props.startTime);
      // end = new Date(this.props.endTime);
      // date = this.getDate(start);
      // startTime = this.getTime(start);
      // endTime = this.getTime(end);
      <View>
              <Text>Group: {item.group}</Text> 
              <Text>Event: {item.name}</Text>
              <Text>Date: {this.getDate(new Date(item.starttime))}</Text>
              <Text testID = 'startTime'>Start Time: {this.getTime(new Date(item.starttime))}</Text>
              <Text testID = 'endTime'>End Time: {this.getTime(new Date(item.endtime))}</Text>
              <Text>Reason: {item.reason}</Text>
              {/* <Text testID = 'conflicts'>Conflicts: {this.props.conflicts}</Text> */}
              <Button handleClick = {() =>this.acceptRequest(item.requestid,this.props.refresh)}
                      label = 'Accept'
                      />
              <Button handleClick = {() => this.denyRequest(item.requestid, this.props.refresh)}
                      label = 'Deny'
                      />
            </View>
    );
    render() { 
        //console.log(this.props);
        //console.log('start time', this.props.startTime);
        let start = new Date(this.props.startTime);
        let end = new Date(this.props.endTime);
        date = this.getDate(start);
        startTime = this.getTime(start);
        endTime = this.getTime(end);
        return (  
            // <View>
            //   <Text>Group: {this.props.group}</Text> 
            //   <Text>Event: {this.props.event}</Text>
            //   <Text>Date: {date}</Text>
            //   <Text testID = 'startTime'>Start Time: {startTime}</Text>
            //   <Text testID = 'endTime'>End Time: {endTime}</Text>
            //   <Text>Reason: {this.props.reason}</Text>
            //   {/* <Text testID = 'conflicts'>Conflicts: {this.props.conflicts}</Text> */}
            //   <Button handleClick = {() =>this.acceptRequest(this.props.id,this.props.refresh)}
            //           label = 'Accept'
            //           />
            //   <Button handleClick = {() => this.denyRequest(this.props.id, this.props.refresh)}
            //           label = 'Deny'
            //           />
            // </View>

            <SafeAreaView>
              <FlatList
                  data = {this.props.roomList}
                  renderItem = {this.renderItem}
                  keyExtractor = {item => item.requestid.toString()}
              />
            </SafeAreaView>
        );
    }
}
 
export default AdminRoomRequest;