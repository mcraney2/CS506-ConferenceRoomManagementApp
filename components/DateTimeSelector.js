import React, {Component} from 'react';
import {View, Platform, Text} from 'react-native';
import DateTimePicker from '@react-native-community/datetimepicker';
import Button from './Button'
class DateTimeSelector extends Component {
    constructor(props) {
        super(props);
        this.state = {  
            //date: new Date(1598051730000),
            mode: 'date',
            show: false
        }
    }

    onChange = (event, selectedDate) => {
        const currentDate = selectedDate || date;
        this.setState({show: Platform.OS === 'ios'});
        //this.setState({date: currentDate});
        this.props.setDate(currentDate);
      };
    showMode = (currentMode) => {
        this.setState({show:true});
        this.setState({mode: currentMode});
      };
    
    showDatepicker = () => {
        this.showMode('date');
      };
    
    showTimepicker = () => {
        this.showMode('time');
      };
      getDate(date) {
          
          
        var year = date.getFullYear();
        var month = date.getMonth() + 1;
        if (parseInt(month, 10) < 10) {
          month = '0' + month;
        }
        var day = date.getDate();
        if (parseInt(day, 10) < 10) {
          day = '0' + day;
        }
       
        var total = month + '/' + day + '/' + year;
        return total;
    }
  getTime(date) {
      
      
      var hour = date.getHours();
      if (parseInt(hour, 10) < 10) {
        hour = '0' + hour;
      }
      var minutes = date.getMinutes();
      if (parseInt(minutes, 10) < 10) {
          minutes = '0' + minutes;
      }
      var total = hour + ':' + minutes;

      return total;
  }
    render() { 
        
        return (  
            <View>
                <View>
                    
                    <Button handleClick={this.showDatepicker} label="Select Date" />
                    <Text>Current Date: {this.getDate(this.props.date)}</Text>
                </View>
                <View>
                    <Button handleClick={this.showTimepicker} label="Select Time" />
                    <Text>Time: {this.getTime(this.props.date)}</Text>
                </View>
                {this.state.show && (
                    <DateTimePicker
                        testID="dateTimePicker"
                        value={this.props.date}
                        mode={this.state.mode}
                        is24Hour={true}
                        display="default"
                        onChange={this.onChange}
                    />
      )}
    </View>
        );
    }
}
 
export default DateTimeSelector;

