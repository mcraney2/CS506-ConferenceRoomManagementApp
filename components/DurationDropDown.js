//import DropDownPicker from 'react-native-dropdown-picker';
import React, { Component } from 'react';
import { StyleSheet, Text, View } from 'react-native';
import {Picker} from '@react-native-picker/picker';

class DurationDropDown extends Component {
    constructor(props) {
        super(props);
        this.state = {  
        
        }
    }
    render() { 
        return (  
            <>
                <View style = {styles.container}>
                
                    <Picker
                        selectedValue = {this.props.hours}
                        onValueChange = {this.props.setHours}
                        testID = 'Hours'
                        style={{ height: 25, width: 150 }}
                        >
                            <Picker.Item label = '0 hours' value = '0' />
                            <Picker.Item label = '1 hour' value = '1' />
                            <Picker.Item label = '2 hours' value = '2' />
                            <Picker.Item label = '3 hours' value = '3' />
                        </Picker>
                    <Picker
                        selectedValue = {this.props.minutes}
                        onValueChange = {this.props.setMinutes}
                        style={{ height: 25, width: 150 }}
                        testID = 'Minutes'
                        >
                            <Picker.Item label = '0 minutes' value = '0' />
                            <Picker.Item label = '15 minutes' value = '15' />
                            <Picker.Item label = '30 minutes' value = '30' />
                            <Picker.Item label = '45 minutes' value = '45' />
                            </Picker>
                    </View>

</>
        );
    }
}
const styles = StyleSheet.create({
    container: {
      flex: 1,
      flexDirection: 'row',
      alignItems: 'center',
      //justifyContent: 'center',
    },
    headerText: {
        fontSize: 22,
        fontWeight: 'bold'
    }
})
export default DurationDropDown;

