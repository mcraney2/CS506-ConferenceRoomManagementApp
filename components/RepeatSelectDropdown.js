//import DropDownPicker from 'react-native-dropdown-picker';
import React, { Component } from 'react';
import { StyleSheet, Text, View } from 'react-native';
import {Picker} from '@react-native-picker/picker';

class RepeatSelectDropdown extends Component {
    constructor(props) {
        super(props);
        this.state = {  
            options : ['None','Weekly', 'Monthly']
        }
    }
    optionsList = () => {
        return( this.state.options.map( (x,i) => { 
            return( <Picker.Item label={x} key={i} value={x}  />)} ));
    }
    
    render() { 
        
        return (  
            <>
                <View style = {styles.container}>
                
                    <Picker
                        selectedValue = {this.props.repeat}
                        onValueChange = {this.props.setRepeat}
                        style={{ height: 25, width: 150 }}
                        >
                            {this.optionsList()}
                        </Picker>
              
                    </View>

            </>
        );
    }
}

const styles = StyleSheet.create({
    container: {
      flex: 1,
      flexDirection: 'row',
      alignItems: 'center',
      //justifyContent: 'center',
    },
    headerText: {
        fontSize: 22,
        fontWeight: 'bold'
    }
})

export default RepeatSelectDropdown;