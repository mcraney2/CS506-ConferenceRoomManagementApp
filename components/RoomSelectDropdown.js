//import DropDownPicker from 'react-native-dropdown-picker';
import React, { Component } from 'react';
import { StyleSheet, Text, View } from 'react-native';
import {Picker} from '@react-native-picker/picker';

class RoomSelectDropdown extends Component {
    constructor(props) {
        super(props);
        this.state = {  
            rooms : ['1080','1100', '1220', '1070' ]
        }
    }
    roomsList = () => {
        return( this.state.rooms.map( (x,i) => { 
            return( <Picker.Item label={x} key={i} value={x}  />)} ));
  }
    
    render() { 
        
        return (  
            <>
                <View style = {styles.container}>
                
                    <Picker
                        selectedValue = {this.props.room}
                        onValueChange = {this.props.setRoom}
                        style={{ height: 25, width: 150 }}
                        >
                            {this.roomsList()}
                        </Picker>
              
                    </View>

</>
        );
    }
}
const styles = StyleSheet.create({
    container: {
      flex: 1,
      flexDirection: 'row',
      alignItems: 'center',
      //justifyContent: 'center',
    },
    headerText: {
        fontSize: 22,
        fontWeight: 'bold'
    }
})
export default RoomSelectDropdown;

