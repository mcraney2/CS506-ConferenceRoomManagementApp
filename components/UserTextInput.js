import React, { Component } from 'react';
import { TextInput } from 'react-native-gesture-handler';

class UserTextInput extends Component {
    constructor(props) {
        super(props);
        this.state = {  
            height: 30,
            
            width: 250,
            borderColor: "grey",
            borderWidth: 1,
            margin: 10,
        }
    }
    updateSize = (height) => {
        this.setState({
          height
        });
      }
    render() { 
        const newValue = this.props.value;
        const height = this.state;
        let newStyle = height;
        return (  
        <TextInput
            placeholder= {this.props.placeHolder}
            onChangeText={(value) => this.props.setValue(value)}
            style={newStyle}
            editable={true}
            multiline={true}
            value={newValue}
            testID = 'text'
            onContentSizeChange={(e) => this.updateSize(e.nativeEvent.contentSize.height)}
          />);
    }
}
 
export default UserTextInput;