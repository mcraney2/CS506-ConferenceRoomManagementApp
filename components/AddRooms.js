import React, { Component } from 'react';
import { Text, TouchableOpacityBase, View,StyleSheet} from 'react-native';

import DateTimeSelector from './DateTimeSelector'
import DurationDropDown from './DurationDropDown'
import RoomSelectDropdown from './RoomSelectDropdown'
import UserTextInput from './UserTextInput'
import Button from './Button'
import axios from 'axios'
import { TextInput } from 'react-native-gesture-handler';

class AddRooms extends Component {
    constructor(props) {
        super(props);
        this.state = {
            roomNum: '0',
        }
    }
    setRoomNum (newRoom) {
        this.setState({roomNum:newRoom});
    }
    sendRequest(room) {
        console.log(room);
        const request = JSON.stringify(
          { 
            roomnumber:room,
        });
        axios.post('http://10.0.2.2:8000/room_mgmt/admin/add_room/', request)
        .then(function (response) {
          console.log(response);
        })
        .catch(function (error) {
          console.log(error);
        });
      }
    render() {
        return (
            <View>
                <UserTextInput 
                    placeHolder = 'Conference Room Name/Number'
                    value = {this.roomNum}
                    setValue = {this.setRoomNum.bind(this)}
                />
                <Button 
                    handleClick={() => this.sendRequest(this.state.roomNum)}
                    label="Add"
                />
            </View>
        );
    }

}

export default AddRooms;