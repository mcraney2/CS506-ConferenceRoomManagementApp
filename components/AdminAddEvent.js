import React, { Component } from 'react';
import { Text, TouchableOpacityBase, View,StyleSheet} from 'react-native';

import DateTimeSelector from '../components/DateTimeSelector'
import DurationDropDown from '../components/DurationDropDown'
import RoomSelectDropdown from '../components/RoomSelectDropdown'
import UserTextInput from '../components/UserTextInput'
import Button from '../components/Button'
import axios from 'axios'
import { TextInput } from 'react-native-gesture-handler';
import RepeatSelectDropdown from './RepeatSelectDropdown';

class AdminAddEvent extends Component {
    
    constructor(props) {
        super(props);
        this.state = { 
            date: new Date(),
            hours: '0',
            minutes: '0',
            room: '0',
            value: '',
            repeat:'none',
            roomList: []
        }
    }
    setDate (newDate) {
        this.setState({date:newDate});
    }

    setMinutes (newMinutes) {
        this.setState({minutes: newMinutes});
    }

    setHours (newHours) {
        this.setState({hours: newHours});
    }

    setRoom (newRoom) {
        this.setState({room : newRoom});
    }

    setValue (newValue) {
        this.setState({value: newValue});
    }

    setRepeat(newRepeat) {
        this.setState({repeat: newRepeat});
    }

    sendRequest(room, startTime, endTime, value, repeat) {
        console.log(room,startTime, endTime, value, repeat.value);
        const request = JSON.stringify(
          { 
            //userid: 1,
            //creator: creator,
            //eventname:event.value,
            eventname: value,
            roomnumber:room,
            starttime:startTime,
            endtime:endTime,
            repeat:repeat.value,

        });
        axios.post('http://10.0.2.2:8000/room_mgmt/admin/events/create/', request)
        .then(function (response) {
          console.log(response);
        })
        .catch(function (error) {
          console.log(error);
        });
    }


    createRoomList(roomslist) {
              let rooms =roomslist;
              let temp = [];
              for (let i = 0; i < rooms.length; i++) {
                  temp.push(rooms[i].roomnumber);
              }
              this.setState({roomList: temp})
    }
    componentDidMount() {
        //console.log("Send database request to get requests")
        this.getRooms();
          
      }
      getRooms(){
        axios.post('http://10.0.2.2:8000/room_mgmt/user/rooms/')
        // fetch('http://10.0.2.2:8000/room_mgmt/user/rooms/', {
        //     method: 'GET',
        //     body: request
        // })
          .then(response => {
              createRoomsList(response.data.roomslist);
          })
          .catch(function(error) {
              console.log(error)
          })
      }
        parseTime(date) {
        var year = date.getFullYear();
        var month = date.getMonth() + 1;
        var day = date.getDate();
        var hour = date.getHours();
        var minutes = date.getMinutes();
        var total = year + '-' + month + '-' + day + ' ' + hour + ':' + minutes;

        return total;
    }

    render() { 
        
        //const endTime = Object.assign({}, this.state.date);
        const currentHour = this.state.date.getHours();
        const currentMinute = this.state.date.getMinutes();
        const endTime = new Date();
        endTime.setHours(currentHour, currentMinute + (60 * parseInt(this.state.hours,10)) + parseInt(this.state.minutes,10), 0, 0);
        //"starttime":"2021-10-25 14:30",
        let startDate = this.parseTime(this.state.date);
        let endDate = this.parseTime(endTime);
        let currentTime = this.parseTime(new Date());
        let creator = "test";
        //console.log(startDate, endDate, currentTime);

        return ( 
            <>
            <View>
                <Text style={styles.textSty}>Add Event To:</Text>
                <RoomSelectDropdown room = {this.state.room} setRoom = {this.setRoom.bind(this)} roomList = {this.state.roomList}/>
                <Text style={styles.textSty3}>Event Name:</Text>
                <UserTextInput placeHolder = 'Enter Event Name Here' value = {this.state.value} setValue = {this.setValue.bind(this)}/>
                <Text style={styles.textSty3}>Date and Time of Event:</Text>
                <DateTimeSelector date = {this.state.date} setDate = {this.setDate.bind(this)}/>
                <Text style={styles.textSty3}>Event Duration:</Text>
                <DurationDropDown minutes = {this.state.minutes} setMinutes = {this.setMinutes.bind(this)} hours = {this.state.hours} setHours = {this.setHours.bind(this)}/>
                <Text style={styles.textSty3}>Repeat:</Text>
                <RepeatSelectDropdown repeat = {this.state.repeat} setRepeat = {this.setRepeat.bind(this)}/>
                <Button 
                        handleClick = {() => this.sendRequest(this.state.room, startDate, endDate, this.state.value, this.state.repeat)}
                        label = "Add Event"
                    />
            </View>
            </>
        );
    }
}
const styles = StyleSheet.create({
    input: {
        paddingRight: 10,
        lineHeight: 23,
        flex: 1,
        textAlignVertical: 'top'
    },
    container: {
        flex: 0.8,
        backgroundColor: '#fff',
        alignItems: 'center',
        justifyContent: 'center',
      },
    container2: {
        margin:10,
        flex:1,
        width:350,
    },
    container3: {
        flex:1,
        width:100,
    },
    container4: {
        flex:1,
        width:130,
    },
    sideBySide: {
        flex:0.3,
        margin:10,
        flexDirection:'row',
    },
    button: {
        margin: 10,
    },
    textSty: {
        fontSize:30,
        fontWeight:'bold',
        marginTop: 20,
    },
    textSty2: {
        fontSize:30,
        fontWeight:'bold',
        marginTop: 20,
        marginLeft: 10,
        color:'red',
    },
    textSty3: {
        fontSize:20,
        fontWeight:'bold',
        marginTop: 10,
        marginRight: 10,
        marginBottom: 10,
        marginLeft: 10,
        color:'#474747',
    },
})

export default AdminAddEvent;