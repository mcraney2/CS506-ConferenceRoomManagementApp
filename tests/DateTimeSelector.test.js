import React from 'react';
import renderer from 'react-test-renderer';
import  DurationDropDown  from '../components/DurationDropDown';
import { fireEvent, render } from '@testing-library/react-native';
import Enzyme, {shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import AdminAddEvent from '../components/AdminAddEvent';
import RoomSelectDropdown from '../components/RoomSelectDropdown';
import UserTextInput from '../components/UserTextInput';
import {setMinutes} from '../components/AdminAddEvent'
import RepeatSelectDropdown from '../components/RepeatSelectDropdown';

import DateTimeSelector from '../components/DateTimeSelector'

test('Renders DateTimeSelector snapshot as expected', () => {
    const tree = renderer.create(<DateTimeSelector/>).toJSON();
    expect(tree).toMatchSnapshot();
});

test('getTime returns correctly formatted minute time', () => {
    Enzyme.configure({adapter: new Adapter()})
    const test = shallow(<DateTimeSelector/>);
    const instance = test.instance();
    const testDate = new Date(2018, 11, 24, 10, 33, 30, 0);
    let result = instance.getTime(testDate);
    expect(result).toBe('10:33');

    const testDate1 = new Date(2018,11, 24, 9, 33, 30, 0);
    let result1 = instance.getTime(testDate1);
    expect(result1).toBe('09:33');

    const testDate2 = new Date(2018,11, 24, 9, 8, 30, 0);
    let result2 = instance.getTime(testDate2);
    expect(result2).toBe('09:08');

    const deadDate = undefined;
    let result3 = instance.getTime(deadDate);
    expect(result3).toBe('');
    
});

test('getDate returns correctly formatted minute time', () => {
    Enzyme.configure({adapter: new Adapter()})
    const test = shallow(<DateTimeSelector/>);
    const instance = test.instance();
    const testDate = new Date(2018, 10, 24, 10, 33, 30, 0);
    let result = instance.getDate(testDate);
    expect(result).toBe('11/24/2018');

    const testDate1 = new Date(2018,8, 24, 9, 33, 30, 0);
    let result1 = instance.getDate(testDate1);
    expect(result1).toBe('09/24/2018');

    const testDate2 = new Date(2018,8, 4, 9, 8, 30, 0);
    let result2 = instance.getDate(testDate2);
    expect(result2).toBe('09/04/2018');

    const deadDate = undefined;
    let result3 = instance.getDate(deadDate);
    expect(result3).toBe('');
    
});

test('All states are initialized correctly', () => {
    Enzyme.configure({adapter: new Adapter()})
    const test = shallow(<DateTimeSelector/>);
    expect(test.state('mode')).toBe('date');
    expect(test.state('show')).toBe(false);
    

  });
  test('Show Mode changes state of show and mode', () => {
    const functionMock = jest.fn();
    Enzyme.configure({adapter: new Adapter()})
    const test = shallow(<DateTimeSelector/>);
    test.showMode = functionMock;
    test.showMode('time');


  });

  test('OnChange changes state of show and mode', () => {
    const functionMock = jest.fn();
    Enzyme.configure({adapter: new Adapter()})
    const test = shallow(<DateTimeSelector/>);
    const testDate2 = new Date(2018,8, 4, 9, 8, 30, 0);
    test.onChange = functionMock;
    test.onChange(testDate2);


  });

  