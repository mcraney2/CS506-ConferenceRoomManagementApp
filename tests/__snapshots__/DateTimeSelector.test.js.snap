// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders DateTimeSelector snapshot as expected 1`] = `
<View>
  <View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "red",
            "borderRadius": 25,
            "justifyContent": "center",
            "paddingHorizontal": 25,
            "paddingVertical": 12,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 18,
            }
          }
        >
          Select Date
        </Text>
      </View>
    </View>
    <Text>
      Current Date: 
      
    </Text>
  </View>
  <View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "red",
            "borderRadius": 25,
            "justifyContent": "center",
            "paddingHorizontal": 25,
            "paddingVertical": 12,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 18,
            }
          }
        >
          Select Time
        </Text>
      </View>
    </View>
    <Text>
      Time: 
      
    </Text>
  </View>
</View>
`;
